support/cas-server-support-aup-ldap/src/main/java/org/apereo/cas/aup/LdapAcceptableUsagePolicyRepository.java

```diff
@@ -52,7 +59,7 @@ public LdapAcceptableUsagePolicyRepository(final TicketRegistrySupport ticketReg
             LdapUtils.LDAP_SEARCH_FILTER_DEFAULT_PARAM_NAME,
             CollectionUtils.wrap(id));
         LOGGER.debug("Constructed LDAP filter [{}]", filter);
-        val connectionFactory = LdapUtils.newLdaptiveConnectionFactory(ldap);
+        val connectionFactory = connectionFactoryList.get(ldap);
```

Comment:
```suggestion
        val connectionFactory = connectionFactoryList.get(ldap.getLdapUrl());
```

Reply:
Do you want to make getLdapUrl as key? If so then I need to rewrite line 62 too.

Reply:
Yes. Using a key based on a properties object here is scary since the object's structure is controlled by lombok. The key would be somewhat "non-deterministic". Using an LDAP url as a String is safer, and possibly slightly faster as a lookup key.